lib/extras/dec/jxl.h

```diff
@@ -55,7 +55,7 @@ struct JXLDecompressParams {
   bool unpremultiply_alpha = false;
 
   // Controls the effective bit depth of the output pixels.
-  JxlBitDepth output_bitdepth = {JXL_BIT_DEPTH_FROM_PIXEL_FORMAT, 0, 0};
+  JxlBitDepth output_bitdepth = {JXL_BIT_DEPTH_FROM_CODESTREAM, 0, 0};
```

Comment:
this would break decoding 12-bit JXLs to PNG, see PR #2619

Reply:
it seems that we always rescale to full 8 or 16 bit in png writer, and that causes the small error in 12-bit pgm -> jxl -> png roundtrip, since it is not possible to exactly represent 12-bit images in 16-bit png, I think it is ok

Reply:
In principle we could use the `sBIT` chunk to indicate that it's actually a 12-bit png and make it roundtrip exactly, even if we do need to pad things to 16-bit. But anyway, that's kind of orthogonal to this PR :)